ClosedXML/Excel/CalcEngine/Functions/Text.cs

```diff
@@ -84,7 +84,22 @@ private static object Concatenate(List<Expression> p)
             foreach (var x in p)
             {
                 if (x is XObjectExpression objectExpression)
-                    throw new CellValueException("This function does not accept cell ranges as parameters.");
+                {
+                    if (objectExpression.Value is CellRangeReference cellRangeReference)
+                    {
+                        if (!cellRangeReference.Range.RangeAddress.IsValid)
+                            throw new CellReferenceException();
+
+                        // Only 1 column per parameter allowed.
+                        var columnCount = cellRangeReference.Range.RangeAddress.LastAddress.ColumnNumber - cellRangeReference.Range.RangeAddress.FirstAddress.ColumnNumber + 1;
+
+                        if (columnCount > 1)
+                            throw new CellValueException("This function does not accept cell ranges as parameters.");
```

Comment:
![image](https://user-images.githubusercontent.com/19576939/62466394-44e5b300-b7a2-11e9-9289-38622228ad4f.png)

Reply:
Oh, so the transpose is also valid. 

In the screenshot you pasted, only E3 and F3 would currently be possble. So what should we do? Either return wrong values or throw an exception?

Reply:
Maybe it is time to implement context-dependent functions? Like `ROW()`, `COLUMN()`, etc.

Reply:
Yes, we should. But for now we need to fix this regression. I think we just return `1 text` or `5 text` for each value in the column/row.

Reply:
I am not convinced that this is a regression as this function never worked with range references as it should. I would prefer the function screaming out loud that it is not fully implemented (=throw an exception) rather than silently returning an _incorrect_ result that might be extremely hard to trace should it cause the error far down the stream.

Reply:
The above use-case isn't a regression, but the `CONCATENATE(A1, B1)` case is a regression.

Yes, I now agree with you. Let's throw an exception. I doubt many people use that edge case anyway.